cpp/cppsdk/gsdk.cpp

```diff
@@ -492,18 +624,19 @@ namespace Microsoft
 
             const std::string GSDK::getSharedContentDirectory()
             {
-                // Declare as static so that it doesn't live on the stack (since we're returning a reference)
-                static const std::string empty = "";
+				auto* gsdk = GSDKInternal::get();
+				if (gsdk == nullptr)
+					return std::string();
 
-				std::lock_guard<std::mutex> lock(GSDKInternal::get().m_configMutex);
+				std::lock_guard<std::mutex> lock(gsdk->m_configMutex);
 
-				const std::unordered_map<std::string, std::string> config = GSDKInternal::get().m_configSettings;
+				const std::unordered_map<std::string, std::string> config = gsdk->m_configSettings;
 
                 auto it = config.find(GSDK::SHARED_CONTENT_FOLDER_KEY);
 
                 if (it == config.end())
                 {
-                    return empty;
+                    return std::string();
```

Comment:
Please restore this stack optimization, as we are very attentive to heap-allocations.